extensions/amp-apester-media/0.1/amp-apester-media.js

```diff
@@ -0,0 +1,141 @@
+  viewportCallback() {
+    // TODO: if amp left the viewport, send iframe post message that it has been seen.
+  }
+
+  /** @override */
+  buildCallback() {
+    /**
+     * @private {?String}
+     */
+    this.displayBaseUrl_ = window.location.protocol +
+      '//stage3-renderer.qmerce.com'; //renderer.qmerce.local.com
+
+    /**
+     * @private {?Element}
+     */
+    this.mediaAttribute_ = user.assert(
+      (this.element.getAttribute('data-apester-media-id') ||
+      this.element.getAttribute('data-apester-channel-token')),
+      'Either the data-apester-media-id or the data-apester-channel-token ' +
+      'attributes must be specified for <amp-apester-media> %s',
+      this.element);
+
+    /**
+     * @private {?Element}
+     */
+    this.iframe_ = null;
+    /**
+     * @private {?Promise}
+     */
+    this.iframePromise_ = null;
+
+    //TODO change to setStyle?
+    this.element.classList.add('-amp-apester-container');
+  }
+
+  /** @override */
+  firstLayoutCompleted() {
+    // Do not hide placeholder
+  }
+
+  /** @override */
+  layoutCallback() {
+    const iframe = this.element.ownerDocument.createElement('iframe');
+    iframe.setAttribute('frameborder', '0');
+    iframe.setAttribute('allowtransparency', 'true');
+    iframe.setAttribute('scrolling', 'no');
+    iframe.src = this.displayBaseUrl_ + '/interaction/'
+      + this.mediaAttribute_;
+    iframe.height = this.element.getAttribute('height');
+    iframe.classList.add('-amp-apester-iframe');
+    this.applyFillContent(iframe);
+
+    setStyles(iframe, {
+      'opacity': 0,
+    });
+
+    // TODO: fallback hide loader if there iframe.src return 404
+    listenFor(iframe, 'embed-size', data => {
+
+      // We only get the message if and when there is a media to display.
+      this.togglePlaceholder(false);
+      let height = data.interaction.data.size.height;
+      const width = data.interaction.data.size.width;
+      height += (data.interaction.layout.directive === 'contest-poll');
+      iframe.height = height;
+      iframe.width = width;
+      const amp = iframe.parentElement;
+      amp.setAttribute('height', height);
+      amp.setAttribute('width', width);
+      this./*OK*/changeHeight(height);
+    }, /* opt_is3P */false);
+
+
+    // append iframe
+    this.element.appendChild(iframe);
+    return this.iframePromise_ = loadPromise(iframe).then(() => {
+      this.getVsync().mutate(() => {
+        setStyles(iframe, {
+          'opacity': 1,
+        });
+      });
+    });
+  }
+
+
+  /** @override */
+  createPlaceholderCallback() {
+    const img = this.element.ownerDocument.createElement('amp-img');
+    const placeholder = this.element.ownerDocument.createElement('div');
+
+    // white background
+    placeholder.setAttribute('placeholder', '');
+    //placeholder.width = this.element.getAttribute('width');
+    placeholder.height = this.element.getAttribute('height');
+    placeholder.className = '-amp-apester-loader-container';
```

Comment:
My request is to exclude placeholder in the first PR and send a separate PR for it. It will be easier to focus on the requirements for placeholder/loader functionality this way. We may, as a result, add some additional provisioning to loaders in the core. In a nutshell, I'd like to discuss whether (a) an SVG can be used instead, and (b) how to start/stop animation when the embed is in/out of viewport. All of these need a separate attention and it'd go faster in a separate PR.